io/net/uperf_test.py

```diff
@@ -0,0 +1,129 @@
+        self.peer_ip = self.params.get("peer_ip", default="")
+        if self.iface not in interfaces:
+            self.cancel("%s interface is not available" % self.iface)
+        if self.peer_ip == "":
+            self.cancel("%s peer machine is not available" % self.peer_ip)
+        self.peer_user = self.params.get("peer_user_name", default="root")
+        uperf_download = self.params.get("uperf_download", default="https:"
+                                         "//github.com/uperf/uperf/"
+                                         "archive/master.zip")
+        tarball = self.fetch_asset(uperf_download, expire='7d')
+        archive.extract(tarball, self.teststmpdir)
+        self.uperf_dir = os.path.join(self.teststmpdir, "uperf-master")
+        cmd = "scp -r %s %s@%s:/tmp" % (self.uperf_dir, self.peer_user,
+                                        self.peer_ip)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to copy the uperf into peer machine")
+        tmp = "cd /tmp/uperf-master;./configure ppc64le;make"
+        cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, tmp)
+        if process.system(cmd, ignore_status=True, shell=True, sudo=True):
+            self.cancel("Unable to compile Uperf into peer machine")
+        os.chdir(self.uperf_dir)
+        process.system('./configure ppc64le', shell=True)
+        build.make(self.uperf_dir)
+        self.uperf = os.path.join(self.uperf_dir, 'doc')
+        self.expected_tp = self.params.get("EXPECTED_THROUGHPUT", default="85")
+        self.uperf_run = str(self.params.get("UPERF_SERVER_RUN", default=0))
+
+    def test(self):
+        """
+        Test run is a One way throughput test. In this test, we have one host
+        transmitting (or receiving) data from a client. This transmit large
+        messages using multiple threads or processes.
+        """
+        if self.uperf_run == '1':
+            tmp = "cd /tmp/uperf-master/src;./uperf -s"
+            cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, tmp)
+            obj = process.SubProcess(cmd, verbose=False, shell=True)
+            obj.start()
+        speed = int(read_file("/sys/class/net/%s/speed" % self.iface))
+        self.expected_tp = int(self.expected_tp) * speed / 100
+        os.chdir(self.uperf)
+        time.sleep(1)
+        cmd = "h=%s proto=tcp uperf -m throughput.xml -a" % self.peer_ip
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status:
+            self.fail("FAIL: Uperf Run failed")
+        for line in result.stdout.splitlines():
+            if self.peer_ip in line:
+                if 'Mb/s' in line:
+                    tput = int(line.split()[3].split('.')[0])
+                else:
+                    # Converting the throughput calculated in Gb to Mb
+                    tput = int(line.split()[3].split('.')[0]) * 1000
```

Comment:
what if through put returned in some other metrics, other than Mbps and Gbps

Reply:
Ethtool and sysfs prints the throughput in MBps. 
Considering throughput for a network card will be in MB or GB, the uperf will also calculate performance in those metrics.  I have no other metric to be considered. 

Also, I've seen a crad operating in KB speed.